pkg/unidling/api/types.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import (
+	kextapi "k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
+	IdledAtAnnotation      = "idling.alpha.openshift.io/idled-at"
+	UnidleTargetAnnotation = "idling.alpha.openshift.io/unidle-targets"
+	NeedPodsReason         = "NeedPods"
+)
+
+// RecordedScaleReference is a SubresourceReference to a scale subresource that also
+// has the previous replica count recorded
+type RecordedScaleReference struct {
+	// TODO: update this to a CrossVersionObjectReference once we pull in a Kube version
+	// with an unversioned apis/autoscaling
+	kextapi.SubresourceReference `json:",inline"`
```

Comment:
Sure, that makes sense.  I don't disagree with what you're saying.   I'm just not overly a fan of doing different things across different parts of the API.  If we have a plan to replace uses of `CrossVersionObjectReference` with `ActuallyCorrectlyReferringToThingsObjectReference`, that's fine, but just having a different way to refer to things for this one piece here seems like it would be confusing, and wouldn't help to clarify the distinction between kind and resource.